### DISCUSSION TITLE
How to keep active websocket connection when the app is in Minimize mode for tablet?

### DISCUSSION THREAD

**Discussion by rahulmishra1991** [microsoft/react-native-windows#12245] [Q&A] [answered]
I am using websocket connection for communicating with my third party app. I am using Ping-Pong mechanism to keep my connection active. But in the tablet when the app goes minimized Ping is stopped working and after 10 minutes automatically websocket connection disconnected. 

Is there any solution so I can send PING even app is in minimized mode.

**Comment by chrisglein** [comment]
This isn't going to be a question specific to React Native. Have you tried looking for answers on UWP for this on stackoverflow or otherwise?

**Reply by rahulmishra1991** [reply]
This is specific question for React Native because when the app is minimized in tablet that time app completely freezes and all the action stopped working till back to maximized and didn't get any solution to keep active app in background for windows app. 

Even I tried to stopping the minimize into tablet but nothing helped for windows app.

I have posted same issue on stackoverflow but didn't get any response yet.

**Reply by chrisglein** [reply]
Can you add a link to the stackoverflow post so we can follow along?
What I mean about not specific to React Native is that RNW sits on top of the native platform to present a cross platform rendering layer and some basic services. There comes a point though where the solution is to drop down into the native code underneath to implement platform-specific customizations. This sounds like it might be one of them. Are you managing this connection purely in the JS or in native code?

**Reply by rahulmishra1991** [reply]
Here is the stackoverflow link:
https://stackoverflow.com/questions/77356885/how-to-keep-active-websocket-connection-when-the-react-native-windows-app-is-in

I am managing connection with JS only.

**Reply by chrisglein** [reply, marked as answer]
I'm not an expert on this topic, but here are some resources I found that may be useful from a native UWP perspective:
- https://learn.microsoft.com/en-us/windows/uwp/networking/network-communications-in-the-background
- https://learn.microsoft.com/en-us/windows/uwp/launch-resume/run-in-the-background-indefinetly
- https://learn.microsoft.com/en-us/uwp/api/windows.networking.sockets.websocketkeepalive?view=winrt-22621
- https://stackoverflow.com/questions/30036377/windows-universal-app-maintain-open-websocket

Obviously the concept of "minimized" isn't a thing on mobile, but apps being shut down while in the background is. Here are some similar queries for mobile scenarios:
- https://stackoverflow.com/questions/40339747/react-native-web-socket-in-background-mode
- https://stackoverflow.com/questions/59616549/websocket-disconnecting-when-app-is-on-background-react-native
- https://blog.logrocket.com/run-react-native-background-tasks-headless-js/

It's by design that resources are reduced for a minimized app, so it's generally not easy to opt out of that behavior. Likely you'll need to manage a background task and/or drop to native code to get the behavior you want. I'm not aware of any out of the box JS solutions that would get you there on their own.
